app/Extensions/Servers/Proxmox/Proxmox.php

```diff
@@ -759,21 +771,25 @@ public function terminateServer($user, $parmas, $order, $product, $configurableO
             'destroy-unreferenced-disks' => 1,
         ];
         $response = $this->deleteRequest('/nodes/' . $parmas['node'] . '/' . $vmType . '/' . $vmid, $postData);
-        if (!$response->json()) throw new Exception('Unable to terminate server');
+        if (!$response->json())
+            throw new Exception('Unable to terminate server');
         return true;
     }
 
 
     public function getCustomPages($user, $parmas, $order, $product, $configurableOptions)
     {
         $vmType = $parmas['type'];
-        $vmid = $parmas['config']['vmid'];
+        $vmid = ExtensionHelper::getOrderProductConfig('vmid', $product->id);
```

Comment:
Add a check if a value is returned

Reply:
check if its null?

Reply:
Yes, In your helper function you return if there is no value, which should also be a `return null;` btw.

Reply:
if ($vmid == null) return;?